src/System.Runtime/tests/System/GCTests.netstandard1.7.cs

```diff
@@ -0,0 +1,184 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Runtime;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Tests
+{
+    public static partial class GCTests
+    {
+        private const long maxGarbage = 1000;
+        public static void MakeSomeGarbage()
+        {
+            Version vt;
+
+            for (int i = 0; i < maxGarbage; i++)
+            {
+                // Create objects and release them to fill up memory
+                // with unused objects.
+                vt = new Version();
+            }
+        }
+        [Fact]
+        public static void GetGeneration_WeakReference()
+        {
+            var myobj = new Version();
+            MakeSomeGarbage();
+            WeakReference wkref = new WeakReference(myobj);
+            var currentgen = GC.GetGeneration(wkref);
+            myobj = null;
+
+            Assert.True(currentgen >= 0);
+            GC.Collect(currentgen, GCCollectionMode.Forced, true, true);
+
+            Assert.Throws<ArgumentNullException>(() => GC.GetGeneration(wkref));
+
+        }
+        [Fact]
+        public static void GCNotifiicationNEgTests()
+        {
+            Assert.Throws<ArgumentOutOfRangeException>(() => GC.RegisterForFullGCNotification(-1, -1));
+            Assert.Throws<ArgumentOutOfRangeException>(() => GC.RegisterForFullGCNotification(100, -1));
+            Assert.Throws<ArgumentOutOfRangeException>(() => GC.RegisterForFullGCNotification(-1, 100));
+
+            Assert.Throws<ArgumentOutOfRangeException>(() => GC.RegisterForFullGCNotification(10, -1));
+            Assert.Throws<ArgumentOutOfRangeException>(() => GC.RegisterForFullGCNotification(-1, 10));
+            Assert.Throws<ArgumentOutOfRangeException>(() => GC.RegisterForFullGCNotification(100, 10));
+            Assert.Throws<ArgumentOutOfRangeException>(() => GC.RegisterForFullGCNotification(10, 100));
+
+
+            Assert.Throws<ArgumentOutOfRangeException>(() => GC.WaitForFullGCApproach(-2));
+            Assert.Throws<ArgumentOutOfRangeException>(() => GC.WaitForFullGCComplete(-2));
+        }
+
+        [Fact]
+        public static void GCNotifiicationTests()
+        {
+            Assert.True(TestWait(true, -1));
+            Assert.True(TestWait(false, -1));
+            Assert.True(TestWait(true, 0));
+            Assert.True(TestWait(false, 0));
+            Assert.True(TestWait(true, 1000));
+            Assert.True(TestWait(false, 1000));
+            Assert.True(TestWait(true, int.MaxValue));
+            Assert.True(TestWait(false, int.MaxValue));
+
+        }
```

Comment:
How long does this test take to run?  It looks from a skim like it could take tens of seconds.  We should find a way to make it take less, and then also make it [OuterLoop].